src/components/common/OldMap.js

```diff
@@ -0,0 +1,722 @@
+    Object.entries(state).map(check => {
+      if (!check[1]) {
+        falseBtn.push(check);
+      }
+    });
+    if (falseBtn.length > 0) {
+      const getBtn = [];
+      falseBtn.map(btn => getBtn.push(btn[0]));
+      setFilterData(
+        data.default.data.filter(
+          i =>
+            !getBtn.includes(
+              i.tags_str.includes(',') ? getFirstType(i.tags_str) : i.tags_str
+            )
+        )
+      );
+    } else {
+      setFilterData(data.default.data);
+    }
+  }, [state]);
+
+  useEffect(() => {
+    const listener = e => {
+      if (e.key === 'Escape') {
+        setSelected(null);
+      }
+    };
+    window.addEventListener('keydown', listener);
+
+    return () => {
+      window.removeEventListener('keydown', listener);
+    };
+  }, []);
+
+  // Potentially pull out into a separate component
+  const typeOfIncidents = data => {
+    console.log('Data points', data);
+    if (data.includes('Presence')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ 'font-size': '2rem', color: 'DarkRed' }}
+          />
+        </div>
+      );
+    } else if (data.includes('Soft')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'Green', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else if (data.includes('Hard')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'Turquoise', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else if (data.includes('Projectiles')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'RoyalBlue', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else if (data.includes('Chemical')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'PaleVioletRed', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else if (data.includes('EnergyDevices')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'Magenta', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else if (data.includes('Miscellaneous')) {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'LightSlateGrey', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    } else {
+      return (
+        <div className="incidents_icons">
+          <i
+            class="fa fa-map-marker"
+            style={{ color: 'Red', 'font-size': '2rem' }}
+          />
+        </div>
+      );
+    }
+  };
```

Comment:
When approaching more than 3 conditions, it might be more performant and easier to maintain this control-flow in a switch statement. Please consider refactoring into a separate file as well.